tools/cli/lib/parseargs.c

```diff
@@ -251,193 +268,108 @@ TDNFCopyOptions(
-    }
-    else if(!strcasecmp(pszName, "skipobsoletes"))
-    {
-        dwError = AddSetOptWithValues(
-                      pCmdArgs,
-                      CMDOPT_KEYVALUE,
-                      pszName,
-                      "1");
-        BAIL_ON_CLI_ERROR(dwError);
-    }
-    else if(!strcasecmp(pszName, "skipsignature"))
-    {
-        dwError = AddSetOptWithValues(
-                      pCmdArgs,
-                      CMDOPT_KEYVALUE,
-                      pszName,
-                      "1");
-        BAIL_ON_CLI_ERROR(dwError);
-    }
-    else if(!strcasecmp(pszName, "skipdigest"))
-    {
-        dwError = AddSetOptWithValues(
-                      pCmdArgs,
-                      CMDOPT_KEYVALUE,
-                      pszName,
-                      "1");
-        BAIL_ON_CLI_ERROR(dwError);
-    }
-    else if(!strcasecmp(pszName, "setopt"))
-    {
-        if(!optarg)
-        {
-            dwError = ERROR_TDNF_CLI_OPTION_ARG_REQUIRED;
-            BAIL_ON_CLI_ERROR(dwError);
+            BAIL_ON_CLI_ERROR((dwError = ERROR_TDNF_CLI_OPTION_ARG_REQUIRED));
         }
+
         dwError = AddSetOpt(pCmdArgs, optarg);
         if (dwError == ERROR_TDNF_SETOPT_NO_EQUALS)
         {
             dwError = ERROR_TDNF_CLI_SETOPT_NO_EQUALS;
         }
-        BAIL_ON_CLI_ERROR(dwError);
     }
-    else if (!strcasecmp(pszName, "noplugins"))
+    else if (!strcasecmp(pszName, "exclude"))
     {
-        dwError = AddSetOptWithValues(
-                      pCmdArgs,
-                      CMDOPT_KEYVALUE,
-                      pszName,
-                      "1");
-        BAIL_ON_CLI_ERROR(dwError);
-    }
-    else if(!strcasecmp(pszName, "enableplugin"))
-    {
-        dwError = AddSetOptWithValues(pCmdArgs,
-                                      CMDOPT_ENABLEPLUGIN,
-                                      pszName,
-                                      optarg);
+        dwError = TDNFAllocateString(pszArg, &pszCopyArgs);
         BAIL_ON_CLI_ERROR(dwError);
+
+        ToFree = pszCopyArgs;
+
+        while ((pszToken = strsep(&pszCopyArgs, ",:")))
+        {
+            dwError = AddSetOptWithValues(pCmdArgs,
+                                CMDOPT_KEYVALUE,
+                                pszName,
+                                pszToken);
+            BAIL_ON_CLI_ERROR((dwError && (pszCopyArgs = ToFree)));
+        }
+
+        pszCopyArgs = ToFree;
     }
-    else if(!strcasecmp(pszName, "disableplugin"))
+    else
     {
-        dwError = AddSetOptWithValues(pCmdArgs,
-                                      CMDOPT_DISABLEPLUGIN,
-                                      pszName,
-                                      optarg);
-        BAIL_ON_CLI_ERROR(dwError);
+        uint32_t i = 0;
+
+        for (i = 0; i < ARRAY_SIZE(OptValTable); i++)
+        {
+            dwError = TDNFAllocateString(OptValTable[i].OptName, &pszCopyArgs);
+            BAIL_ON_CLI_ERROR(dwError);
+
+            ToFree = pszCopyArgs;
```

Comment:
We need to retain the starting address of pszCopyArgs because strsep will modify it's address